### ISSUE TITLE
Allow optional unix sockets to speed up requests on a single machine

### ISSUE THREAD

**Issue by bloof-bb** [uavaustin/orchestra#14]
This will help quite a bit for services that either have to get their data in a time-sensitive manner, or can help with the imagery service so that we can send images quickly without having to deal with working with volumes (and reimplementing the imagery storage system in other services).

We could use a common mounted directory in `/var/run/sockets/` (or something like that) per machine.

When making this kind of change, we'd need to actually add the protocols in the URLs so we know what we're using exactly. Something like `unix:///var/run/sockets/whatever.sock` or `unix+http:///var/run/sockets/whatever.sock` could work for the HTTP over unix sockets. (The later format is used in [this python requests library](https://github.com/msabramo/requests-unixsocket).) The rest of the URLs would need to start with `http://` as they probably should already.

Note that the services hosting APIs should still keep serving their content over TCP as they are currently, the unix sockets would just be extra.

**Comment by oldmud0** [comment]
Yeah, I was wondering why we were not using Unix sockets. But I am not sure if the latency benefits are significant in Docker. A benchmark would be nice if I could find one.

**Comment by bloof-bb** [comment]
As far I as I know the only web frameworks we're using are express.js, which I know has support for http over unix sockets, and then we have Phoenix, which I'm not finding good documentation on how to do it for that.
